enhancements/sdk-generate-commands.md

```diff
@@ -0,0 +1,353 @@
+---
+title: README describing new command workflows for OLM enabled operators
+authors:
+  - "@varshaprasad96"
+  - "@estroz"
+reviewers:
+  - "@jlanford"
+  - "@dmesser"
+  - "@jmrodri"
+approvers:
+  - "@jlanford"
+  - "@dmesser"
+  - "@jmrodri"
+creation-date: 2020-04-09
+last-updated: 2020-06-02
+status: implementable
+see-also:
+  - https://github.com/operator-framework/operator-sdk/pull/2860
+---
+
+# New `generate` command workflows for OLM enabled operators
+
+## Release Signoff Checklist
+
+- \[x\] Enhancement is `implementable`
+- \[x\] Design details are appropriately documented from clear requirements
+- \[x\] Test plan is defined
+- \[ \] Graduation criteria for dev preview, tech preview, GA
+- \[ \] User-facing documentation is created.
+
+## Open Questions:
+
+* _(Old project layouts)_ Should a user be able to generate both a bundle and
+package manifests format in the same directory (for migration purposes), or should
+they be mutually exclusive, ex. `generate bundle` will return an error if a
+`generate packagemanifests` has been run in the same output directory, and visa versa?
```

Comment:
Are they different sets of files? If they can co-exist and work in OLM then I think it's okay. But it does seem confusing that we would allow this. I'd rather see it error if one exists. And give me an option `--force` to blow away anything that is there before. 

Here is how I see it working from a cli perspective.

1. `generate packagemanifests` creates the package manifest files
2. `generate bundle` this will error out because the package manifest already exists.

Now if I want to force the issue I'd run the following:

1. `generate packagemanifests` creates the package manifest files
2. `generate bundle --force` removes the package manifest files and lays down a bundle.

Also I see it both ways.

Reply:
Right now both a bundle and packages are generated in `deploy/olm-catalog/<operator-name>`, so you could have:
```
deploy/olm-catalog/memcached-operator/
├── memcached-operator.package.yaml
├── 0.0.1
│   ├── cache.example.com_memcacheds_crd.yaml
│   └── memcached-operator.v0.0.1.clusterserviceversion.yaml
├── manifests
│   ├── cache.example.com_memcacheds_crd.yaml
│   └── memcached-operator.clusterserviceversion.yaml
└── metadata
    └── annotations.yaml
```
Which can confuse some bundle vs. package manfiests detection logic in the SDK. It would be preferable to keep them separate at least. `--force` might be nice for migration purposes.

Reply:
Thanks.